src/typechecker.erl

```diff
@@ -4165,87 +4240,95 @@ add_type_pat(Pat, ?type(union, UnionTys)=UnionTy, Env, VEnv) ->
         _SomeTysMatched ->
             %% TODO: The constraints should be merged with *or* semantics
             %%       and var binds with intersection
-	    {Ty, Cs} = glb(PatTys, Env),
+            {Ty, Cs} = glb(PatTys, Env),
+            NewEnv = union_var_binds(Envs, Env),
             {Ty,
              normalize(type(union, UBounds), Env),
-             union_var_binds(VEnvs, Env),
+             NewEnv,
```

Comment:
Why this change?

Reply:
Probably just a debugging artefact. Reverting.